controller/instance_manager_controller.go

```diff
@@ -1118,8 +1138,8 @@ func (m *InstanceManagerMonitor) Run() {
 			m.lock.Unlock()
 
 			if needUpdate {
```

Comment:
consider inverting this `if !needUpdate { continue }`
reduces nesting.

Reply:
Thank you. Fixed